modules/data.remote/inst/SMAPvsModel_Comparison.R

```diff
@@ -0,0 +1,353 @@
+
+## CONUS NEON Sites XML File Location ##
+input_dir <- '~/pecan/modules/data.remote/inst/conus_sites.xml'
+
+## File Directory to Place Output PDFs In ##
+output_dir <- '/projectnb/dietzelab/jbowers1/graphic_output/'
+
+## Output Directories for geoJson Files ##
+geoJSON_outdir <- '/projectnb/dietzelab/jbowers1/geoFiles/'
+
+## Output Directories for SMAP netCDF Files ##
+smap_outdir <- '/projectnb/dietzelab/jbowers1/smap_ncFiles/'
+
+## Download SMAP data ##
+source('~/pecan/modules/data.remote/inst/download_SMAP.R')
+all_smap_data <- download_SMAP(start, end, geoJSON_outdir, smap_outdir)
+####################^ INPUTS ^####################
+start <- as.Date(start)
+end <- as.Date(end)
+
+
+#' @title extractSoilMoist
+#' @description extracts a single ensamble member during the given year range from a specified file location
+#'
+#' @param betySiteId bety site id for the desired location (String)
+#' @param yearStart starting year YYYY as (date) or (numeric)
+#' @param yearEnd ending year YYYY as (date) or (numeric)
+#' @param ENS ensamble member front padded with 0's to five values (String)
+#'
+#' @return A list containing soil moisture and time values
+#'
+#' @examples 
+#' extractSoilMoist(betySiteId = '1000004945', yearStart = 2016, yearEnd = 2017, ENS = '25')
+#' 
+#' This extracts soil moisture values and time values from the 25th ensamble member netCDF file 
+#' in a specified file directory for the Harvard Forest ('1000004945') site during the years 
+#' 2016 and 2017.
+#' 
+extractSoilMoist <- function(betySiteId, yearStart, yearEnd, ENS){
+
+  for (x in 0:(yearEnd - yearStart)) {
+    nc.file <- paste0(ensamble_dir, 'ENS-', ENS, '-', betySiteId, '/', yearStart + x,'.nc')
+    nc <- nc_open(nc.file)
+    if (x == 0) {
+      soilMoist <- ncvar_get(nc,'SoilMoist')
+      time <- PEcAn.utils::cf2datetime(nc[["dim"]][["time"]][["vals"]],nc$dim$time$units)
+    } else {
+      soilMoist <- append(soilMoist, ncvar_get(nc,'SoilMoist'))
+      time <- append(time, PEcAn.utils::cf2datetime(nc[["dim"]][["time"]][["vals"]],nc$dim$time$units))
+    }
+  }
+  return(list(time, soilMoist))
+}
+
+
+#' @title extractEnsamlbes
+#' @description utilizes extractSoilMoist function to extract an entire ensamble of a specified
+#' over the course of a specified date range.
+#'
+#' @param betySiteId bety site id for the desired location as (String)
+#' @param ENSnum number of members in the ensamble as (numeric)
+#'
+#' @return a data frame consisting of the soil mositure as well as their date-time values 
+#' during the specified date range
+#' 
+#'
+#' @examples 
+#' extractEnsambles(betySiteId = '1000004945', start = as.Date('2016-01-02'), end = as.Date('2016-07-01'))
+#' 
+#' Extracts all 25 ensample members (default) from the Harvard Forest (1000004945) site between the
+#' '2016-01-02' - '2016-07-01' date range.
+#' 
+extractEnsambles <- function(betySiteId, ENSnum=25){
```

Comment:
would suggest a rename to make it clear this function just grabs soil moisture, not other variables. That said, it wouldn't take much work in either function to make the variable to be extracted an argument rather than something that's hardcoded

Reply:
Renamed to extractVariableENS.